tests/e2e/test_rosacli_upgrade.go

```diff
@@ -449,6 +507,81 @@ var _ = Describe("Describe/List rosa upgrade",
 			})
 	})
 
+// Version struct to hold version information
+type Version struct {
+	Original string
+	Parts    []int
+}
+
+// ParseVersion parses a version string into a Version struct
+func ParseVersion(version string) Version {
+	parts := strings.Split(version, ".")
+	intParts := make([]int, len(parts))
+	for i, part := range parts {
+		if num, err := strconv.Atoi(part); err == nil {
+			intParts[i] = num
+		} else {
+			intParts[i] = 0
+		}
+	}
+	return Version{
+		Original: version,
+		Parts:    intParts,
+	}
+}
+
+func (v Version) Less(u Version) bool {
+	for i := 0; i < len(v.Parts) && i < len(u.Parts); i++ {
+		if v.Parts[i] != u.Parts[i] {
+			return v.Parts[i] < u.Parts[i]
+		}
+	}
+	return len(v.Parts) < len(u.Parts)
+}
+
+// VersionSlice attaches the methods of sort.Interface to []Version
+type VersionSlice []Version
+
+func (s VersionSlice) Len() int           { return len(s) }
+func (s VersionSlice) Less(i, j int) bool { return s[i].Less(s[j]) }
+func (s VersionSlice) Swap(i, j int)      { s[i], s[j] = s[j], s[i] }
+
+// SortVersions sorts a slice of version strings
+func SortVersions(versions []string) []string {
+	parsedVersions := make([]Version, len(versions))
+	for i, v := range versions {
+		parsedVersions[i] = ParseVersion(v)
+	}
+	sort.Sort(sort.Reverse(VersionSlice(parsedVersions)))
+	sortedVersions := make([]string, len(versions))
+	for i, v := range parsedVersions {
+		sortedVersions[i] = v.Original
+	}
+	return sortedVersions
+}
```

Comment:
We have had `sort` functions https://github.com/openshift/rosa/blob/master/tests/utils/exec/rosacli/version_service.go#L221 for this. You just need to use it directly in `FindUpperYStreamVersion`.